OpenLIFUTransducerTracker/OpenLIFUTransducerTracker.py

```diff
@@ -2121,6 +2212,156 @@ def run_fiducial_registration(self,
+        transformType: int = 1,
+        numLandmarks: int = 200,
+        numIterations: int = 100
+    ):
+        """Registers a moving model to a fixed model using the Iterative Closest Point (ICP) algorithm.
+        Note: This function operates directly on the point sets of the
+        input models and does not consider any parent transforms. Therefore,
+        both input models should be defined within the coordinate system intended for registration.
+
+        Args:
+            input_fixed_model (vtkMRMLModelNode): The fixed model (target) to which the moving model will be registered.
+            input_moving_model (vtkMRMLModelNode): The moving model (source) that will be transformed to align with the fixed model.
+            transformType (int, optional): The type of transformation to be estimated.
+                - 0: Rigid body transformation
+                - 1: Similarity transformation
+                - 2: Affine transformation 
+                Defaults to 1 (Similarity).
+            numLandmarks: Maximum number of landmarks sampled from the moving model. The default is 200.
+            numIterations: Maximum number of iterations. The default is 100.
+
+        Returns:
+            vtkMRMLTransformNode or None: A new vtkMRMLTransformNode containing the computed
+            transformation that aligns the moving model with the fixed model. Returns None
+            if the registration fails. The transform node will be automatically added to the
+            scene.
+        """
+        
+        icp_result_node =  slicer.mrmlScene.AddNewNodeByClass("vtkMRMLLinearTransformNode", "icp_transform_result")
+
+        try:
+            icpTransform = vtk.vtkIterativeClosestPointTransform()
+            icpTransform.SetSource( input_moving_model.GetPolyData() )
+            icpTransform.SetTarget( input_fixed_model.GetPolyData() )
+            icpTransform.GetLandmarkTransform().SetModeToRigidBody()
+            if transformType == 1:
+                icpTransform.GetLandmarkTransform().SetModeToSimilarity()
+            if transformType == 2:
+                icpTransform.GetLandmarkTransform().SetModeToAffine()
+            icpTransform.SetMaximumNumberOfIterations( numIterations )
+            icpTransform.SetMaximumNumberOfLandmarks( numLandmarks )
+            icpTransform.Modified()
+            icpTransform.Update()
+
+            icp_result_node.SetMatrixTransformToParent( icpTransform.GetMatrix() )
+
+            if slicer.app.majorVersion >= 5 or (slicer.app.majorVersion >= 4 and slicer.app.minorVersion >= 11):
+                icp_result_node.SetNodeReferenceID(slicer.vtkMRMLTransformNode.GetMovingNodeReferenceRole(), input_moving_model.GetID())
+                icp_result_node.SetNodeReferenceID(slicer.vtkMRMLTransformNode.GetFixedNodeReferenceRole(), input_fixed_model.GetID())
+
+        except Exception as e:
+            print("Exception thrown:", e)
+            return None
```

Comment:
Arbitrary exceptions get swallowed up here and reduced to a print statement, but then in places where `run_icp_model_registration` is used we do not check if `None` was returned and do error handling.

I think it would be better to allow the exception to be raised, or re-raise it here. And ensure that if ICP fails then there is a reasonable user-facing outcome, not just a traceback or a print in the console. Could be as simple as an error dialog saying "ICP failed, check the logs."